### ISSUE TITLE
from_yaml filter doesn't evaluate jinja expressions in read variables

### ISSUE THREAD

**Issue by techzilla** [ansible/ansible#81535]
### Summary

When using the from_yaml filter on content, provided by the lookup file plugin, jinja expressions are not evaluated.   Instead the value is a string containing the jinja expression. 

### Issue Type

Bug Report

### Ansible Version

```console
ansible [core 2.14.9]
  config file = /home/username/projects/ansible-skel/ansible.cfg
  configured module search path = ['/home/username/projects/ansible-skel/library']
  ansible python module location = /home/username/projects/ansible-skel/.venv/lib/python3.10/site-packages/ansible
  ansible collection location = /mnt/c/Users/username/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/username/projects/ansible-skel/.venv/bin/ansible
  python version = 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] (/home/username/projects/ansible-skel/.venv/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
ANSIBLE_NOCOWS(/home/username/projects/ansible-skel/ansible.cfg) = True
CONFIG_FILE() = /home/username/projects/ansible-skel/ansible.cfg
DEFAULT_ASK_PASS(/home/username/projects/ansible-skel/ansible.cfg) = False
DEFAULT_GATHERING(/home/username/projects/ansible-skel/ansible.cfg) = smart
DEFAULT_HOST_LIST(/home/username/projects/ansible-skel/ansible.cfg) = ['/home/username/projects/ansible-skel/inventory']
DEFAULT_MODULE_PATH(/home/username/projects/ansible-skel/ansible.cfg) = ['/home/username/projects/ansible-skel/library']
DEFAULT_NO_TARGET_SYSLOG(/home/username/projects/ansible-skel/ansible.cfg) = True
DEFAULT_ROLES_PATH(/home/username/projects/ansible-skel/ansible.cfg) = ['/home/username/projects/ansible-skel/required-roles', >
DEFAULT_TIMEOUT(/home/username/projects/ansible-skel/ansible.cfg) = 20
HOST_KEY_CHECKING(/home/username/projects/ansible-skel/ansible.cfg) = False
RETRY_FILES_ENABLED(/home/username/projects/ansible-skel/ansible.cfg) = False

CONNECTION:
==========

paramiko_ssh:
____________
host_key_checking(/home/username/projects/ansible-skel/ansible.cfg) = False

ssh:
___
host_key_checking(/home/username/projects/ansible-skel/ansible.cfg) = False
timeout(/home/username/projects/ansible-skel/ansible.cfg) = 20
```

### OS / Environment

WSL, Windows 10.
Also confirmed on a recently updated OSX. 

### Steps to Reproduce

content of playbook,
```yaml 
- name: Debug read_yaml
  hosts: localhost
  gather_facts: false

  vars:
    file: "{{ playbook_dir }}/test.yaml"
    test: "{{ lookup('file', file) | from_yaml }}"

  tasks:

  - debug:
      var: test

  - debug:
      msg: "{{ item }}"
    loop: "{{ test }}"

```
content of file named test.yaml, also in playbook directory
```yaml
- metadata:
    name: test
  version: "{{ ansible_version }}"
```

### Expected Results

```
PLAY [Debug read_yaml] ***********************************************************************************

TASK [debug] *********************************************************************************************
ok: [localhost] => {
    "test": [
        {
            "metadata": {
                "name": "test"
            },
            "version": "{{ ansible_version }}"
        }
    ]
}

TASK [debug] *********************************************************************************************
ok: [localhost] => (item={'metadata': {'name': 'test'}, 'version': {'string': '2.10.17', 'full': '2.10.17', 'major': 2, 'minor': 10, 'revision': 17}}) => {
    "msg": {
        "metadata": {
            "name": "test"
        },
        "version": {
            "full": "2.10.17",
            "major": 2,
            "minor": 10,
            "revision": 17,
            "string": "2.10.17"
        }
    }
}

PLAY RECAP ***********************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

This is the output from Ansible version 2.10.4

### Actual Results

```console
PLAY [Debug read_yaml] ***************************************************************************************************

TASK [debug] *************************************************************************************************************
ok: [localhost] => {
    "test": [
        {
            "metadata": {
                "name": "test"
            },
            "version": "{{ ansible_version }}"
        }
    ]
}

TASK [debug] *************************************************************************************************************
ok: [localhost] => (item={'metadata': {'name': 'test'}, 'version': '{{ ansible_version }}'}) => {
    "msg": {
        "metadata": {
            "name": "test"
        },
        "version": "{{ ansible_version }}"
    }
}

PLAY RECAP ***************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by s-hertel** [comment]
Hi!

Thanks very much for your submission to Ansible.  It means a lot to us that you've taken time to contribute.

Unfortunately, we're not sure if we want this feature in the program, and I don't want this to seem confrontational.
Our reasons for this are:

  * This is working as designed, lookups are unsafe. You can accomplish this by using the `template` lookup.

However, we're absolutely always up for discussion.
Because this project is very active, we're unlikely to see comments made on closed tickets and we lock them after some time.
If you or anyone else has any further questions, please let us know by using any of the communication methods listed in the page below:

   * https://docs.ansible.com/ansible/latest/community/communication.html

In the future, sometimes starting a discussion on the development list prior to implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

**Comment by techzilla** [comment]
@s-hertel This isn't a feature, this worked in version 2.10.4, it doesn't work any longer, this is not something new. Please re-open the ticket, or conversely document that a specific change caused playbooks that worked to no longer work and that was intentional.

**Comment by mkrizek** [comment]
> @s-hertel This isn't a feature...

This has always been a feature.

> ... this worked in version 2.10.4, it doesn't work any longer, this is not something new. Please re-open the ticket as a bug, or conversely document that a specific change caused playbooks that worked to no longer work and that was an intentional design change.

From https://docs.ansible.com/ansible/latest/plugins/lookup.html:
> By default, lookup return values are marked as unsafe for security reasons.  If you trust the outside source your lookup accesses, pass allow_unsafe=True to allow Jinja2 templates to evaluate lookup values.

The fact that it "worked" in any version was a security issue. It was fixed in https://github.com/ansible/ansible/pull/77609 and included in `2.12.6`.

> I previously could not use template, because the content would interpolate two times, once when the template lookup occurred and again when the variables were called. The consequence was that I had to write jinja expressions that escaped content like this, {{ '{{ "{{ mustashe_variable }}" }}' }}, which was the reason this was preferable to taking output from template and reading it via from_yaml. Is that something that can be done now? I had no way to know, the problem isn't the design choices Ansible makes, it's that a change of significance was not documented in anyway.

As @s-hertel mentioned above, using the `template` lookup results in the same output as listed in "expected results":
```yaml
- hosts: localhost                                                                                                                                                                                                                                                                                                    
  gather_facts: false
  tasks:
    - debug:
        msg: "{{ lookup('template', playbook_dir ~ '/81535-test.yml') | from_yaml }}"
```

```yaml
- metadata:                                                                                                                                                
    name: test                                                                                                                                             
  version: {{ ansible_version }}
```

```
PLAY [localhost] *****************************************************************************************************************************************************************************************************************************************************************************************************

TASK [debug] *********************************************************************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": [
        {
            "metadata": {
                "name": "test"
            },
            "version": {
                "full": "2.16.0.dev0",
                "major": 2,
                "minor": 16,
                "revision": 0,
                "string": "2.16.0.dev0"
            }
        }
    ]
}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
